samples/multi-thread/wasm-apps/main.c

```diff
@@ -45,23 +48,39 @@ main(int argc, char *argv[])
         goto fail1;
     }
 
+    // O_CREAT on linux (glibc)
+    if (!(sem = sem_open("testsem", 0100))) {
+        printf("Failed to open sem. %p\n", sem);
+        goto fail2;
+    }
+
     pthread_mutex_lock(&mutex);
     if (pthread_create(&tid, NULL, thread, &num) != 0) {
         printf("Failed to create thread.\n");
-        goto fail2;
+        goto fail3;
```

Comment:
Should pthread_mutex_unlock(&mutex) before goto fail3, it is an older bug, could you please help fix it?

Reply:
OK